index.html

```diff
@@ -808,6 +811,90 @@ <h3>
           </p>
         </section>
       </section>
+      <section>
+        <h3>
+          `id` member
+        </h3>
+        <p>
+          The [=manifest's=] <code><dfn data-export="" data-dfn-for=
+          "manifest" id="identity">id</dfn></code> member is a <a>string</a> that
+          represents the <dfn>identity</dfn> for the application. 
+          The |identity| takes the form of a URL, which is same origin as the
+          start URL.
+        </p>
+        <p>The |identity| is used by user agents to uniquely identify the
+        application universally. When the user agent sees a manifest with an
+        |identity| that does not correspond to an already-installed application,
+        it SHOULD treat that manifest as a description of a distinct
+        application, even if it is served from the same URL as that of another
+        application. When the user agent sees a manifest with an |identity|
+        matching that of an already-installed application, it SHOULD be used as
+        a signal that this manifest is a replacement for the already-installed
+        application's manifest, and not a distinct application, even if it is
+        served from a different URL than the one seen previously.
+        </p>
+        <p>
+          The |identity| MAY be used by a service that collects lists of web
+          applications to uniquely identify applications.
+        </p>
+        <p>
+          To <dfn>process the `id` member</dfn>, given [=object=]
+          |json:JSON|, [=ordered map=] |manifest:ordered map|:
+        </p>
+        <ol class="algorithm">
+          <li>Set |manifest|["id"] to |manifest|["start_url"].
```

Comment:
I think we might have another issue here: `start_url` might be `undefined`. 

We would need to assure that `start_url` is always assured for this to work. In that case, the document's URL would again need to be the `start_url`.

Reply:
Yes - this was a bug I noticed recently w/ Chromium & our parsing, and the spec. if start_url is "", then it just isn't set. I am planning on creating a pull request to set start_url to the document url if it is "" - which I believe is the only time it isn't set?

I guess if it isn't in the json - we would want to catch that case here. Perhaps we have no id though - if there is no start_url, there kind of isn't a webapp, just page metadata. So maybe we say id is undefined if start_url is processed to undefined?

Reply:
If we allow start_url to always default to document url, then it could lead to confusion if a developer attaches a manifest on multiple pages (and they don't specify start_url), as, with the `id` member, this could mean each page could trigger an web app update to make that page now the new start_url of the app.

I guess my vote would be to make the behavior more predictable / less potentially weird  by requiring the start_url to be defined for the id to be considered? WDYT @philloooo?

Reply:
#670 changes the default value to manifest's base path. I think alternatively I can update the start_url processing algorithm here to just describe what's happending in all user agents now. Which is do 
```
Set |manifest|["start_ur"] to |document URL:URL| 
``` 
in the beginning of the processing alrogithm, to make it clear that when it's undefined, default to document_url
WDYT?

Reply:
I agree we should resolve this before submitting this pull request. I think there are a couple ideas that @marcoscaceres  is playing with which are looking good. (#670 and #991)

Reply:
oh nice, #991 looks like what I was thinking